### ISSUE TITLE
Windows build problem

### ISSUE THREAD

**Issue by BrianHanke** [AcademySoftwareFoundation/xstudio#57]
Howdy all, I made it fairly far trying to build xStudio on Windows 10, but media_reader_ffmpeg throws about 1000 errors in Visual Studio. Stuff like:

> namespace "xstudio::utility" has no member "sys_time_point"

I installed ffmpeg using vcpkg and pointed FFMPEG_ROOT to C:\source\xstudio-windows\build\vcpkg_installed\x64-windows.

Any suggestions how to fix this? Thanks!

**Comment by nyue** [comment]
@BrianHanke how did you get past the ALSA requirements ?

**Comment by BrianHanke** [comment]
@nyue I don't know what that is, so I'm not sure I did get past! I just followed the Windows build guide. I already have a VM set up with VS 2019, Qt and others, so it was pretty. All went well in CMake GUI, it's just building the solution in VS that fails on media_reader_ffmpeg.

**Comment by nyue** [comment]
@BrianHanke Thanks for the breadcrumb. Are you able to share the URL to the Windows build guide ? The repo has build guides for Centos, Rocky but not Windows

**Comment by BrianHanke** [comment]
@nyue Sure, here you go:

Windows code: https://github.com/mpkepic/xstudio/tree/windows
Build guide: https://github.com/mpkepic/xstudio/blob/windows/docs/build_guides/windows.md

**Comment by RdMaple** [comment]
visual studio can't  build,many c++ error.

**Comment by nyue** [comment]
I am attempting to build xstudio for windows from the source and avoiding running the powershell script and vcpkg, my workplace windows workstation is severely locked down

I am now at the stage for missing ALSA as a dependency. What open source project can I build that can provide this ALSA (audio/sound ?) package on Windows ?

Cheers

**Comment by nyue** [comment]
I get the following opencolorio/vcpkg error, any idea ?

I know there is some minimum cmake version, is there some minimum git version required ?

```
-- VCPKG: install from manifest; using target triplet: x64-windows
error: failed to execute: "C:\Program Files\Git\cmd\git.exe" "--git-dir=C:\Users\yueni\projects\xstudio\build\vcpkg\.git" "--work-tree=C:\Users\yueni\projects\xstudio\build\vcpkg\buildtrees\versioning_\versions\opencolorio\9569944b76966b78bec5ef83672899acd7e4febe_544.tmp" -c core.autocrlf=false read-tree -m -u 9569944b76966b78bec5ef83672899acd7e4febe
error: git failed with exit code: (128).
fatal: failed to unpack tree object 9569944b76966b78bec5ef83672899acd7e4febe
note: while checking out port opencolorio with git tree 9569944b76966b78bec5ef83672899acd7e4febe
CMake Error at cmake/modules/vcpkg.cmake:477 (message):
  VCPKG: install from manifest failed
Call Stack (most recent call first):
  cmake/modules/vcpkg.cmake:236 (vcpkg_install_manifest)
  cmake/modules/vcpkg.cmake:614 (vcpkg_init)
  CMakeLists.txt:146 (include)


-- Configuring incomplete, errors occurred!
```

**Comment by akiraendo1** [comment]
I get the error:
CMake Error at CMakeLists.txt:157 (message):
  Failed to ensurepip.

**Comment by nyue** [comment]
I have some naive questions.

1. Why are we leveraging vcpkg, powershell (from the ASWF discussion, specific powershell version) ?
2. Is it practical to build the individual dependencies e.g. ocio, separately and references them in xStudio build via CMake config i.e. OCIO_DIR ?

My concern is that if we use some high level packaging e.g. vcpkg, when things breaks e.g. hash mismatch, missing files, it is more challenging for contributors to dive in to help (in the ASWF discussion, someone suggested updating some JSON file to point to a different version of opencolorio because the packaging database has missing entries)

**Comment by AlsonEntuna** [comment]
hi, I'm also encountering issues with building the windows version. Whenever I try to configure cmake, it gives me this json issue.
![image](https://github.com/AcademySoftwareFoundation/xstudio/assets/7828218/257630b1-7ace-4d91-835f-143266491e79)
What seems to the be problem? I already have evrything in my machine.

Oh btw, whenever you clone the repository, the `scripts/setup` folder is not there. Plus when executing the powershell script also gives errors on windows
![image](https://github.com/AcademySoftwareFoundation/xstudio/assets/7828218/6da15a62-dd51-4036-b6f7-8b82d6eabb2e)

**Comment by cedricp** [comment]
Hi,
I'm having trouble executing the compiled (with VS 2022) version (after some hours of tweaks!)
I've launched the package_xstudio.bat to copy dependencies, but it looks that something's wrong with python install, the modules aren't found. I've been able to almost fix it by playing with PYTHONHOME/PYTHONPATH env vars, but I still can't open an image. 
I also compiled it on a Linux Box, it's far easier to build, but sandbox install doesn't work. I can run it from the cmake build dir, though.
Best regards.

**Comment by RdMaple** [comment]
I have build Sucess!Make sure

**Comment by RdMaple** [comment]
![xstudio](https://github.com/AcademySoftwareFoundation/xstudio/assets/18696718/334a9529-58d6-4cb0-9557-b4427b203bf3)
Sucess build😁, but cann't play any video 😂!
I will write how i build, please wait for a moment!

**Comment by ronnyascencio** [comment]
I am having problems with :
CMake Error at cmake/modules/vcpkg.cmake:477 (message):
  VCPKG: install from manifest failed
Call Stack (most recent call first):
  cmake/modules/vcpkg.cmake:236 (vcpkg_install_manifest)
  cmake/modules/vcpkg.cmake:614 (vcpkg_init)
  CMakeLists.txt:146 (include)

any Idea?

**Comment by RdMaple** [comment]
1,make sure you are use vs2019(vs2022 not work),and vs language change to english!
2,if you get cmake error,please see log info ,if you get version not found,please modify vcpkg.json in \xstudio\.
3,then you should build sucessful!
4,when you exec package_xstudio.bat,you also need :

**Comment by AlsonEntuna** [comment]
ok apparently I saw the issue on my side because it was on a different branch when you check it out from github.

now that I have the proper branch, I tried and now have this one `failed to ensurepip` thing. I have pip installed but don't know what's wrong with it

```
The system cannot find the file specified


CMake Error at CMakeLists.txt:157 (message):
  Failed to ensurepip.


Configuring incomplete, errors occurred!
```

**Comment by cedricp** [comment]
Ok, I installed VS2019, built the sources, launched the "package_xstudio.bat" and copied all dll files as you told me, but I'm still in trouble with python :
```Python path configuration:
  PYTHONHOME = (not set)
  PYTHONPATH = (not set)
  program name = 'python'
  isolated = 0
  environment = 1
  user site = 0
  import site = 1
  sys._base_executable = 'C:\\Users\\cedric\\Documents\\workspace\\xstudio\\build\\bin\\Release\\xstudio.exe'
  sys.base_prefix = ''
  sys.base_exec_prefix = ''
  sys.platlibdir = 'lib'
  sys.executable = 'C:\\Users\\cedric\\Documents\\workspace\\xstudio\\build\\bin\\Release\\xstudio.exe'
  sys.prefix = ''
  sys.exec_prefix = ''
  sys.path = [
    'C:\\Users\\cedric\\Documents\\workspace\\xstudio\\build\\bin\\Release\\python310.zip',
    '.\\DLLs',
    '.\\lib',
    'C:\\Users\\cedric\\Documents\\workspace\\xstudio\\build\\bin\\Release',
  ]
[2023-11-06 09:58:32.451] [xstudio] [warning] void __cdecl xstudio::embedded_python::EmbeddedPython::setup(void) Failed to setup API : failed to get the Python codec of the filesystem encoding
AttributeError: module 'sys' has no attribute 'stdout'
[2023-11-06 09:58:32.451] [xstudio] [warning] bool __cdecl xstudio::embedded_python::EmbeddedPythonActor::act::<lambda_3>::operator ()(struct caf::connect_atom,const int) const Python error AttributeError: module 'sys' has no attribute 'stdout'
[2023-11-06 09:58:32.451] [xstudio] [warning] Connected failed false
[2023-11-06 09:58:35.536] [xstudio] [warning] void __cdecl xstudio::ui::viewport::Viewport::get_colour_pipeline(void) error("create_colour_pipeline failed, invalid colour pipeline name.")
[2023-11-06 09:58:35.543] [xstudio] [warning] void __cdecl xstudio::ui::viewport::Viewport::get_colour_pipeline(void) error("create_colour_pipeline failed, invalid colour pipeline name.")
[2023-11-06 09:58:37.916] [xstudio] [info] XStudio UI launched.
```
Something's still missing :(

**Comment by ewerybody** [comment]
Hey @AlsonEntuna! :)

> Oh btw, whenever you clone the repository, the `scripts/setup` folder is not there.

You need to turn it to the **windows** branch!

> Plus when executing the powershell script also gives errors on windows

Yeah! I had the same issue when I just downloaded the `.zip` but after I cloned it and switched to windows branch it worked!

Now I'm having other issues 😅 ...

**Comment by AlsonEntuna** [comment]
@ewerybody yeah already got that figured out but the only issue that i have right now that's preventing me from compiling it is the ensurepip cmake macro. Everything else is fine already.

**Comment by ewerybody** [comment]
Yeah I have that as well. I'm left with:
```
Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
no such file or directory

CMake Error at CMakeLists.txt:157 (message):
  Failed to ensurepip.


Configuring incomplete, errors occurred!
```
but eehhh WHAT directory?!?!? 🤷

**Comment by AlsonEntuna** [comment]
@ewerybody , I only have the ensure pip issue right now, I did check my python env and pip is also installed, but when I took a look at the cmake macro it doesn't point me to anything that's why I'm confused on how to solve this issue,

**Comment by AlsonEntuna** [comment]
hello,

So I already managed to compile and package the tool but when launching it, it gives me errors on QT platform initialization. Has any encountered this before?

![qt_issue](https://github.com/AcademySoftwareFoundation/xstudio/assets/7828218/ea384cc0-833b-420f-adae-57c5ad5ea29b)

Is there anything I need to configure first before running it?

**Comment by AlsonEntuna** [comment]
> hello,
> 
> So I already managed to compile and package the tool but when launching it, it gives me errors on QT platform initialization. Has any encountered this before?
> 
> ![qt_issue](https://user-images.githubusercontent.com/7828218/284046873-ea384cc0-833b-420f-adae-57c5ad5ea29b.png)
> 
> Is there anything I need to configure first before running it?

nvm, I already solved it. All good on my end now!

**Comment by ewerybody** [comment]
how did you solve it?

**Comment by OlivierArgentieri** [comment]
I got the same problem as @AlsonEntuna and i have figure out by copying the 'platforms' folder from my qt installation:
`C:\Qt\5.15.2\msvc2019_64\plugins`
and paste it beside my binary, so here in my case `D:\xstudio\prefix_install\bin\Release`

Maybe a better solution exist, maybe an evironment variable that we can set to QT install folder, i don't know but for now it works with the platforms folder and i can go forward to the next error !

**Comment by AlsonEntuna** [comment]
also, the `.bat` file doesn't really copy the necessary .dll files. I had to write a simple python script to copy the dlls that the exe needs

**Comment by OlivierArgentieri** [comment]
Yes, i needed also to copy all dll files manually, same for qt dependencies

**Comment by apanner** [comment]
```
"C:\xstudioBuild\\build\src" does not exist.

"C:\xstudioBuild\\build\bin\Debug\xstudio.exe" does not exist.

"C:\xstudioBuild\\build\bin\Release\xstudio.exe" does not exist.

"C:\xstudioBuild\\build\src" does not exist.

DLL files from "Debug" folders copied to "C:\xstudioBuild\\build\bin\Debug"
DLL files from "Release" folders copied to "C:\xstudioBuild\\build\bin\Release"
Press any key to continue . . .

```
xstudio.exe not created.. after
Visual studio 2019 -> build solution.- so many errors.

**Comment by scottmcdonnell** [comment]
I got the build to work with a ffmpeg 6.1 binary, I haven't tried the vcpkg ffmpeg yet. But same as a previous comment, the player is blank and will not show any video/image. Has anyone succesfully built and does your build load media?

**Comment by fjallraven** [comment]
Same here, built successfully (ffmpeg 5.1, matched all versions same as Linux build) but cannot display video or images

**Comment by mpkepic** [comment]
Just wanted to follow up and point everyone to the latest commit of the windows PR;, the build system improvements addresses a lot of the comments found here; cmake install is now complete and the install script that used to be there is gone.  Plugins and xstudio_root paths get set properly so the plugins that allow you to actually load media (as well as others) can be found etc.

It can now use Ninja and Visual Studio 2022; and comes with presets and a reworked build instructions to make everything super easy.

We've removed the choco script, as we've been able to get the manual build dependencies down to just a couple.

If you are still interested, I implore you to try the latest commit.

Thanks!

**Comment by 7VincentGasti7** [comment]
> Just wanted to follow up and point everyone to the latest commit of the windows PR;, the build system improvements addresses a lot of the comments found here; cmake install is now complete and the install script that used to be there is gone. Plugins and xstudio_root paths get set properly so the plugins that allow you to actually load media (as well as others) can be found etc.
> 
> It can now use Ninja and Visual Studio 2022; and comes with presets and a reworked build instructions to make everything super easy.
> 
> We've removed the choco script, as we've been able to get the manual build dependencies down to just a couple.
> 
> If you are still interested, I implore you to try the latest commit.
> 
> Thanks!

Thank you for your hard work ! It works well ! :)

**Comment by mpkepic** [comment]
Aw thank you!

I'm going to ask if we could close this issue out and open individual issues as they are discovered so we can track each separately.

Cheers!

**Comment by BrianHanke** [comment]
I'm the OP, feel free to close this.

**Comment by jonbragado** [comment]
For those of us that are not programmers, would it be possible to get an install wrapper where we double click and installs on windows? 
As artists we´d love to be able to test this tool but the barrier is simply too high right now. 
Much appreciated!

**Comment by mpkepic** [comment]
Prebuilt binaries are a topic being discussed in ASWF but there are
barriers to be worked out in providing them.  That doesn't mean it won't
happen, just needs figuring out.  It is a topic of heavy concern to many of
us in the ASWF ORI group.  Cheers.

**Comment by jonbragado** [comment]
Thank you for the comment. 
While those discussions happen, the reality is that only those with enough programming and coding experience can access it. Being a tool that is not meant to be used by coders but by artists, it seems a bit difficult to understand the current barrier of having a beta version that users can benefit from installing and trying the software. 

How about somebody who has managed to build a working windows version can upload a drive link for those that we would want to test it at is current state?
